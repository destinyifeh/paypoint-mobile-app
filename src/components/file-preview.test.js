'use strict';

import React from 'react';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import FilePreview from './file-preview';

configure({
  adapter: new Adapter()
});

it('FilePreview With placeholder renders correctly', () => {
  const tree = renderer
    .create(<FilePreview 
      attachment={Object}
      name="File X"
      placeholder="File Placeholder"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('FilePreview Without placeholder renders correctly', () => {
  const tree = renderer
    .create(<FilePreview 
      attachment={Object}
      name="File Y"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<FilePreview 
      attachment={Object}
      placeholder="File Placeholder"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
