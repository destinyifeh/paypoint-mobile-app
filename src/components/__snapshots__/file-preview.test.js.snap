// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePreview With placeholder renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 75,
          "marginRight": 10,
          "width": 75,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#808593",
          "fontSize": 16,
        }
      }
      title={true}
    >
      File X
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 0,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`FilePreview Without placeholder renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 75,
          "marginRight": 10,
          "width": 75,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#808593",
          "fontSize": 16,
          "width": "70%",
        }
      }
      title={true}
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 0,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 75,
          "marginRight": 10,
          "width": 75,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#808593",
          "fontSize": 16,
        }
      }
      title={true}
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 0,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
